Content.Server/Atmos/GasSprayerComponent.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.Interfaces;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.Components.Pointing;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+
+
+namespace Content.Server.Atmos
+{
+    [RegisterComponent]
+    public class GasSprayerComponent : Component, IAfterInteract
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager = default!;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager = default!;
+#pragma warning restore 649
+
+        //TODO: create a function that can create a gas based on a solution mix
+        public override string Name => "GasSprayer";
+
+        private string _spraySound;
+        private string _sprayType;
+        private string _fuelType;
+        private string _fuelName;
+        private int _fuelCost;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _spraySound, "spraySound", string.Empty);
+            serializer.DataField(ref _sprayType, "sprayType", string.Empty);
+            serializer.DataField(ref _fuelType, "fuelType", string.Empty);
+            serializer.DataField(ref _fuelName, "fuelName", "fuel");
+            serializer.DataField(ref _fuelCost, "fuelCost", 50);
+        }
+
+        public void AfterInteract(AfterInteractEventArgs eventArgs)
+        {
+            if (!Owner.TryGetComponent(out SolutionComponent tank))
+                return;
+
+            if (tank.Solution.GetReagentQuantity(_fuelType) == 0)
+            {
+                _notifyManager.PopupMessage(Owner, eventArgs.User,
+                    Loc.GetString("The {0} is out of {1}!", Owner.Name, _fuelName));
```

Comment:
Using the text macro here would be better. They are found in https://github.com/space-wizards/RobustToolbox/blob/58b6e1803060adf51b084b1c5aceff180ee7cd06/Robust.Shared/Localization/Macros/English/EnglishMacros.cs

```suggestion
                    Loc.GetString("{0:theName} is out of {1}!", Owner.Name, _fuelName));
```